### PR TITLE
[Documentation][Book] Search

### PR DESCRIPTION
**Pull request by TheMadeleine**
| Q               | A
| --------------- | ---
| Doc fix?        | no |
| New docs?    | yes |
| Related tickets |  |
| License         | MIT |

### CODE REVIEW

File: docs/book/products/search.rst [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+If you would like to extend this built-in functionality read
+:doc:`this article about grids customizations </customization/grids>`, and :doc:`the GridBundle docs </bundles/SyliusGridBundle/index>`.
+
+ElasticSearch
+-------------
+
+When the grids filtering is not enough for you, and your needs are more complex you should go for the
+`ElasticSearch <https://www.elastic.co/products/elasticsearch>`_ integration.
+
+Sylius is provided with the `Lakion/SyliusElasticSearchBundle <https://github.com/Lakion/SyliusElasticSearchBundle>`_ integration.
```
Comment (lchrusciel) [inline docs/book/products/search.rst]: `Sylius is provided with the` it is not totally true. `You can extend your Sylius with external plugin which will provide  ElasticSearch functionality`?

File: docs/book/products/search.rst [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    filters:
+        search:
+            type: string
+            label: false
+            options:
+                fields: [translation.name]
+            form_options:
+                type: contains
+
+It searches by product names that contain a string that the user typed in.
```
Comment (lchrusciel) [inline docs/book/products/search.rst]: I still don't know how to use it. Do I need to send a request? What should it contain?

File: docs/book/products/search.rst [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+======
+
+Having a products search functionality in an ecommerce system is a very popular usecase. Sylius provides a products search functionality
+that is a grid filter.
+
+Grid filters
+------------
+
+For simple usecases of products search use the **filters of grids**.
+The products grid in the shop, for echa category of products has already a ``search`` filter:
```
Comment (venyii) [inline docs/book/products/search.rst]: `echa` -> `each`

Maybe instead:

For example, the shop's categories each have a `search` filter in the products grid:

File: docs/book/products/search.rst [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+
+The search bar looks like below:
+
+.. image:: ../../_images/search.png
+    :align: center
+
+Customizing search filter
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The search bar in many shops should be more sophisticated, than just a simple text search. You may be needing to add
```
Comment (venyii) [inline docs/book/products/search.rst]: You may need to filter by price, reviews, ...?

File: docs/book/products/search.rst [hunk 5]
```diff
@@ -0,0 +1,61 @@
+.. index::
+   single: Search
+
+Search
+======
+
+Having a products search functionality in an ecommerce system is a very popular usecase. Sylius provides a products search functionality
+that is a grid filter.
```
Comment (michalmarcinkowski) [inline docs/book/products/search.rst]: Redundant blank line

**Comment by pjedrzejewski** [comment]
Thanks Magda!

**Pull request merged**